atef/widgets/utils.py

```diff
@@ -4,10 +4,13 @@
 from typing import ClassVar, Optional
 
 from qtpy import QtCore, QtGui, QtWidgets
-from qtpy.QtCore import QEvent, QObject
-from qtpy.QtGui import QPalette
+from qtpy.QtCore import QEvent, QObject, QRegularExpression
+from qtpy.QtGui import QPalette, QRegularExpressionValidator
 from qtpy.QtWidgets import QLineEdit
 
+PV_regexp = QRegularExpression(r'^\w+(:\w+)+(\.\w+)*$')
```

Comment:
This is partly LCLS-specific (the `:` delimiter, however common it may be even elsewhere)
And also partially incorrect: the field at the end can include JSON or a trailing `$` indicating long string support:
* `PV:NAME.FLD$` (long string)
* `PV:NAME.FLD.{"f":{"lo":0,"hi":10}}` ([channel filter](https://epics.anl.gov/base/R3-15/1-docs/filters.html))
* `PV:NAME.FLD.[1:2]` channel filter array slice shortcut

The TL;DR is that I think regexes for PVs are bound to be wrong, so the best we can/should do is just accept whatever the user types, strip off trailing spaces, and let the control system deal with it.

Reply:
... aw darn.  I should have known better than to expect that I knew what PVs looked like.  The JSON suffix is  fascinating...

Reply:
Can PVs have spaces at all?  I think the validator can still be helpful in that context

Reply:
I've never seen a PV with a space in it, but that's likely just to make terminal input less awful

Reply:
I'll propose `^\S*$` as the new validator, but will of course wait for Ken to chime in

Reply:
The PV part shouldn't have spaces but I'm not sure that applies to the post-`.` JSON blob

Reply:
I've removed the validation and added a bunch of `.strip()` calls.  I left the validator "installed" on the PV line edits in the off chance we want to enforce some validation later.  But maybe mostly because of laziness.